internal/pkg/deploy/upload/asset/matcher.go

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
are we just moving all the match-related logic to a new file?

Reply:
yeah 😊

Reply:
poor danny had to answer the same question twice hahahahaha

Reply:
i probably should have mentioned that somewhere lol